server/plugin/command.go

```diff
@@ -265,6 +271,50 @@ func (p *Plugin) handleSubscriptionsList(_ *plugin.Context, args *model.CommandA
 	return txt
 }
 
+func (p *Plugin) getWebhookListForRepoOrOrg(githubClient *github.Client, repo, owner string, ctx context.Context) (bool, error) {
+	isWebhook := false
+	opt := &github.ListOptions{
+		PerPage: GithubListOptionsPerPageValue,
+	}
+
+	for {
+		var githubHooks []*github.Hook
+		var githubResponse *github.Response
+		var err error
+
+		if repo == "" {
+			githubHooks, githubResponse, err = githubClient.Organizations.ListHooks(ctx, owner, opt)
+		} else {
+			githubHooks, githubResponse, err = githubClient.Repositories.ListHooks(ctx, owner, repo, opt)
+		}
+
+		if err != nil {
+			p.API.LogWarn("Not able to get the list of webhooks", "Owner", owner, "Repo", repo, "Error", err.Error())
+			// Breaking from the loop if the repo or org is not found
+			if strings.Contains(err.Error(), "404 Not Found") {
+				isWebhook = true
+				break
```

Comment:
Breaking here de-facto means returning, right? Why is the 404 case special?

Reply:
404 case is special because we are checking whether there is an org or repo present with the name we put during subscribing if not we don't need any additional messages.